### PR TITLE
tests: context: convert legacy test to ztest

### PR DESCRIPTION
**Pull request by punitvara**
Use ztest apis in legacy test to support ztest
framework.

Signed-off-by: Punit Vara <punit.vara@intel.com>

### CODE REVIEW

**Comment by punitvara** [comment]
@SavinayDharmappa Can you please review ?

File: tests/kernel/context/src/context.c [hunk 1]
```diff
@@ -934,46 +928,38 @@ void main(void)
...
 #ifndef CONFIG_ARM
 	TC_PRINT("Testing k_cpu_atomic_idle()\n");
 	rv = test_kernel_cpu_idle(1);
-	if (rv != TC_PASS) {
-		goto tests_done;
-	}
+	zassert_equal(rv, TC_PASS, " failure kernel cpu idle");
 #endif
 #endif
+	k_thread_priority_set(k_current_get(), old_prio);
```
Comment (andrewboie) [inline tests/kernel/context/src/context.c]: why is this needed?
ztest creates a new thread for every case so I don't see the point

look at tests/ztest/src/ztest.c run_test()

Comment (punitvara) [reply tests/kernel/context/src/context.c]: yes I know that ztest creates a thread with priority -1. I tried to preserve priority after setting it to 0 as it was done in other test cases. But I don't mind removing it.

**Comment by punitvara** [comment]
@andrewboie Can you please review this once ?

**Review by SavinayDharmappa — APPROVED** [review]

**Review by tyrantlizard77 — APPROVED** [review]
